Tests/SwiftParserTest/translated/InitDeinitTests.swift

```diff
@@ -129,7 +129,7 @@ final class InitDeinitTests: ParserTestCase {
       }
       """,
       diagnostics: [
-        DiagnosticSpec(message: "deinitializers cannot have a name", fixIts: ["remove 'x'"])
+        DiagnosticSpec(message: "deinitializers cannot have token", fixIts: ["remove 'x'"])
```

Comment:
This is a down side, as we only have a TokenSyntax here. 

Could not quickly think of a solution that would be more informative. 
Maybe we could add an enum with error types, to the new error. 
And then give it as params instead of the nodes. But that is a bit of anti pattern of what we do other places? 

What do you think @ahoppen